server/controllers/userController.js

```diff
@@ -14,23 +14,47 @@ async function postUsers (req, res, next) {
 	});
 }
 
-async function postToken (req, res, next) {
+async function postTokens (req, res, next) {
 	const { username, password } = req.body;
-	userManager.verifyUser(username, password).then((valid) => {
-		if (valid) {
-			const accessToken = jwt.sign({ username }, process.env.JWT_SECRET);
-			res.cookie('token', accessToken, { overwrite: true });
-			res.cookie('username', username, { overwrite: true });
-			res.sendStatus(200);
-		} else {
-			res.sendStatus(400);
-		}
-	}).catch((e) => {
-		next(e);
-	});
+	const valid = await userManager.verifyUser(username, password);
+	if (valid) {
+		const accessToken = jwt.sign({ username }, process.env.ACCESS_JWT_SECRET, { expiresIn: process.env.TOKEN_EXPIRATION_TIME });
+		const refreshToken = jwt.sign({ username }, process.env.REFRESH_JWT_SECRET);
+
+		res.cookie('accessToken', accessToken, { overwrite: true });
+		res.cookie('refreshToken', refreshToken, { overwrite: true });
+		res.cookie('username', username, { overwrite: true });
```

Comment:
I'm just curious why do you choose to use cookies? And what other approaches here could be?

Reply:
So the other approach would probably be to use local storage or session storage. The reason I chose to use cookies was because for both local and session storage, you cannot access or set them through the server. I would have to be sending extra token data back and forth in post requests in places where I could just use get requests. A drawback with cookies is that they can only store very small data but that shouldn't matter here as I am not storing much.  Additionally, these methods all differ in how long they store their data. For session storage, it stores it until the tab is closed and for local storage, it is stored till deleted. With cookies, you can specify the lifetime so that's also nice, although I am not using that feature here.

Reply:
You're totally right about all vulnerabilities. So it is fine to use cookies if you're checked the approach.
But just an alternative option could be HTTP headers sent with the response from the server. I'm curious have you consider them?

Reply:
I think the way that would be done is with a Token header. However that has the same issue as localStorage and sessionStorage that you need to include them explicitly in any request that requires the token. Maybe that's a good thing though...